library/CMakeLists.txt

```diff
@@ -1,10 +1,3 @@
-# Set the project root directory if it's not already defined, as may happen if
-# the library folder is included directly by a parent project, without
-# including the top level CMakeLists.txt.
-if(NOT DEFINED MBEDTLS_DIR)
```

Comment:
It is not the only thing that would miss if the library project was included directly in a parent project thus the error message wouldn't be very helpful I'd say. It is just something that we do not support thus it can fail in many ways and many other ways in the future.